### PR TITLE
[21407] Remove example

### PR DESCRIPTION
**Pull request by elianalf**
## Description

This PR removes the discovery-server.xsd (because not used) and the example (because there is already one in Fast DDS)

## Contributor Checklist

- [x] Commit messages follow the project guidelines. 
- N/A The added tests pass locally.
- N/A Changes are backwards compatible. 
- N/A New feature has been documented/Current behavior is correctly described in the documentation.
- N/A Applicable backports have been included in the description.

## Reviewer Checklist

- [x] The PR has a milestone assigned.
- [x] The title and description correctly express the PR's purpose.
- [x] Check contributor checklist is correct.
- [x] Check CI results: changes do not issue any warning.
- [x] Check CI results: failing tests are unrelated with the changes.

### CODE REVIEW

File: .github/workflows/config/build.meta [hunk 1]
```diff
@@ -3,3 +3,6 @@ names:
         cmake-args:
             - "-DNO_TLS=ON"
             - "-DSECURITY=OFF"
+    discovery-server:
+        cmake-args:
+            - "-DCOMPILE_EXAMPLES=ON"
```
Comment (JesusPoderoso) [inline .github/workflows/config/build.meta:8]: EOF

File: resources/xsd/discovery-server.xsd [hunk 2]
```diff
@@ -201,7 +201,7 @@
     <xs:complexType name="writerTimesType">
         <xs:all minOccurs="0">
             <xs:element name="initialHeartbeatDelay" type="durationType" minOccurs="0"/>
-            <xs:element name="heartbeatPeriod" type="durationType" minOccurs="0"/>
+            <xs:element name="heartbeat_period" type="durationType" minOccurs="0"/>
```
Comment (JesusPoderoso) [inline resources/xsd/discovery-server.xsd:204]: I am not sure if we should change the XML schema

**Review by JesusPoderoso — CHANGES_REQUESTED** [review]

File: resources/xsd/discovery-server.xsd (hunk 2 above)
Comment (elianalf) [reply resources/xsd/discovery-server.xsd:204]: After internal discussion we decided to remove the file because it is not used

**Review by JesusPoderoso — APPROVED** [review]
LGTM with green CI

**Review by JesusPoderoso — CHANGES_REQUESTED** [review]
Consider removing also [this section](https://github.com/eProsima/Discovery-Server/blob/9e01e11f56a3064005395c7a1256ae12d627883f/CMakeLists.txt#L388) and [this usage](https://github.com/eProsima/Discovery-Server/blob/9e01e11f56a3064005395c7a1256ae12d627883f/CMakeLists.txt#L413) from the main CMakeLists.txt

**Review by JesusPoderoso — APPROVED** [review]
LGTM with green CI

**Pull request merged**